Source/DafnyDriver/LegacyCliCompilation.cs

```diff
@@ -35,14 +35,23 @@ namespace Microsoft.Dafny {
   /// 
   /// Will be replaced by CompilationManager
   /// </summary>
-  public class CompilerDriver : IDisposable {
+  public class LegacyCliCompilation : IDisposable {
     private readonly ExecutionEngine engine;
 
-    public CompilerDriver(DafnyOptions dafnyOptions) {
+    public LegacyCliCompilation(DafnyOptions dafnyOptions) {
       engine = ExecutionEngine.CreateWithoutSharedCache(dafnyOptions);
     }
 
     public static async Task<int> Run(DafnyOptions options) {
+
+      if (options.DafnyProject == null) {
+        var uri = options.CliRootSourceUris.FirstOrDefault() ?? new Uri(Directory.GetCurrentDirectory());
+        options.DafnyProject = new DafnyProject {
+          Includes = Array.Empty<string>(),
+          Uri = uri,
+          ImplicitFromCli = true
+        };
+      }
```

Comment:
Didn't follow why this became necessary

Reply:
Seems like its not. Might have been defensive coding on my part. I'll remove it.